src/components/Editor/Runners/PythonRunner/SkulptRunner/SkulptRunner.jsx

```diff
@@ -71,16 +73,18 @@ const SkulptRunner = ({ active, outputPanels = ["text", "visual"] }) => {
   );
   const codeRunStopped = useSelector((state) => state.editor.codeRunStopped);
   const drawTriggered = useSelector((state) => state.editor.drawTriggered);
+  const senseHatAlwaysEnabled = useSelector(
+    (state) => state.editor.senseHatAlwaysEnabled,
+  );
   const reactAppApiEndpoint = useSelector((s) => s.editor.reactAppApiEndpoint);
   const output = useRef();
   const dispatch = useDispatch();
   const { t } = useTranslation();
   const settings = useContext(SettingsContext);
   const isMobile = useMediaQuery({ query: MOBILE_MEDIA_QUERY });
 
-  const [hasVisualOutput, setHasVisualOutput] = useState(true);
-
-  const [showRunner, setShowRunner] = useState(active);
+  const [hasVisualOutput, setHasVisualOutput] = useState(senseHatAlwaysEnabled);
+  const [visualOutput, setVisualOutput] = useState(true);
```

Comment:
I'm not quite following the logic behind having both of these defined? might be worth considering tweaking the names for more clarity?

Reply:
ah this is an awkward one to rename 😅 essentially `hasVisualOutput` is dynamic based on the code (so it'll change when a visual library is imported) whereas `visualOutput` is toggled on code run and determines what's rendered. The reason I separated it is because only sense HAT (AstroPi wc) initially renders with a visual output tab, all others are inferred from code 🤔 I'll have a think of state names but also spotted a little bug

Reply:
it's fine, I just found it a little confusing to read, but it's not worth going to great lengths for if it's a pain

Reply:
I ended up renaming to `codeHasVisualOutput` and `showVisualOutput` instead 😄 hopefully that helps